storage/interceptors/smaap_service_binding_interceptor.go

```diff
@@ -238,27 +241,54 @@ func (i *ServiceBindingInterceptor) AroundTxDelete(f storage.InterceptDeleteArou
 			return fmt.Errorf("deletion of multiple bindings is not supported")
 		}
 
+		operation, found := opcontext.Get(ctx)
+		if !found {
+			return fmt.Errorf("operation missing from context")
+		}
+
 		if bindings.Len() != 0 {
 			binding := bindings.ItemAt(0).(*types.ServiceBinding)
-
-			operation, found := opcontext.Get(ctx)
-			if !found {
-				return fmt.Errorf("operation missing from context")
+			if operation.Type == types.CREATE && !operation.Context.Async {
+				if err := f(ctx, deletionCriteria...); err != nil {
+					return err
+				}
 			}
-
 			if err := i.deleteSingleBinding(ctx, binding, operation); err != nil {
 				return err
 			}
+		} else if operation.InOrphanMitigationState() && operation.Context.ServiceInstanceID != "" {
+			// In case we don't have instance & use the operation context to perform orphan mitigation
+			binding := types.ServiceBinding{}
+			binding.ServiceInstanceID = operation.Context.ServiceInstanceID
+			binding.ID = operation.ResourceID
+			if err := i.deleteSingleBinding(ctx, &binding, operation); err != nil {
+				return err
```

Comment:
you can't return here err
otherwise if operation failed it will return as error and you won't reach the rest of the code